### PR TITLE
Migration/androidx

### PR DESCRIPTION
**Pull request by philips77**
* All modules migrated to Android Jetpack.
* Sample app migrated to MaterialComponents theme.
* Tabs -> Spaces in sample app.
Tested and seems to work.

### CODE REVIEW

**Comment by philips77** [comment]
Perhaps you could tag and release a version before merging? Can migrated libraries be used by non-migrated projects?

**Comment by bgiori** [comment]
Hmmm, ok maybe not yet. 

I glanced on the PR and it seemed like it was mostly changes to the sample app. If this affects core, I will have to look it over again. 

It looks like the only change to `ble` and `core` is for annotations and test framework? I don't think that warrants breaking backwards compatibility.

**Comment by philips77** [comment]
In the core library it affected only test dependencies, but as tests are absent, it could be commented out for now.

**Comment by philips77** [comment]
I have removed dependencies to android tests in core module. Perhaps the proguard-android file could also be removed.

File: mcumgr-core/build.gradle [hunk 1]
```diff
@@ -38,9 +37,7 @@ dependencies {
     implementation 'com.fasterxml.jackson.core:jackson-databind:2.9.6'
 
     // Test
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
-    androidTestImplementation 'com.android.support.test:runner:1.0.2'
     testImplementation 'junit:junit:4.12'
 }
 
-apply from: rootProject.file('gradle/gradle-mvn-push.gradle')
+// apply from: rootProject.file('gradle/gradle-mvn-push.gradle')
```
Comment (bgiori) [inline mcumgr-core/build.gradle]: Why comment this out?

File: mcumgr-ble/build.gradle [hunk 2]
```diff
@@ -29,8 +29,8 @@ dependencies {
...
     testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'com.android.support.test:runner:1.0.2'
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    androidTestImplementation 'androidx.test:runner:1.1.1'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.1'
 }
 
-apply from: rootProject.file('gradle/gradle-mvn-push.gradle')
\ No newline at end of file
+// apply from: rootProject.file('gradle/gradle-mvn-push.gradle')
```
Comment (bgiori) [inline mcumgr-ble/build.gradle]: why are we commenting this out?

**Comment by philips77** [comment]
Sorry, I'm adding the lib as a module in nRF Connect and Gradle didn't like this line. Pushed by mistake.

File: mcumgr-core/build.gradle (hunk 1 above)
Comment (philips77) [reply mcumgr-core/build.gradle]: Fixedl

File: mcumgr-ble/build.gradle (hunk 2 above)
Comment (philips77) [reply mcumgr-ble/build.gradle]: Fixed.

**Comment by philips77** [comment]
We could wait with releasing this a bit. First other PRs, then another release with those changes. Just to avoid problems with android.support vs androidx.

**Comment by sssemil** [comment]
Is this PR ready for merging?

**Comment by bgiori** [comment]
> Is this PR ready for merging?

I would be ok with merging this in now that androidx has been out for some time. 

@philips77 are you available or would you like me to take this

**Comment by philips77** [comment]
Go ahead. I'm busy these days. Thanks!

**Comment by sssemil** [comment]
Is this still being worked on?

**Comment by bgiori** [comment]
@sssemil Sorry this fell off my radar. Do you need a new version built?

**Comment by sssemil** [comment]
@bgiori no worries :) That would be great.

**Pull request merged**